app/javascript/app/document-capture/components/document-capture.jsx

```diff
@@ -1,30 +1,39 @@
-import React, { useState } from 'react';
+import React, { useState, useContext } from 'react';
 import FormSteps from './form-steps';
 import DocumentsStep, { isValid as isDocumentsStepValid } from './documents-step';
 import SelfieStep, { isValid as isSelfieStepValid } from './selfie-step';
+import MobileIntroStep from './mobile-intro-step';
+import DeviceContext from '../context/device';
 import Submission from './submission';
 
 function DocumentCapture() {
   const [formValues, setFormValues] = useState(null);
+  const { isMobile } = useContext(DeviceContext);
+
+  const steps = [
+    {
+      name: 'documents',
+      component: DocumentsStep,
+      isValid: isDocumentsStepValid,
+    },
+    {
+      name: 'selfie',
+      component: SelfieStep,
+      isValid: isSelfieStepValid,
+    },
+  ];
+
+  if (isMobile) {
+    steps.unshift({
```

Comment:
I always forget what `unshift` does... what if we did the same boolean + filter trick for `steps`? Then it makes it easier to see them all in order?

```js
const steps = [
  isMobile && {
    name: 'intro',
    component: MobileIntroStep,
  },
  {
    // ...
  } 
].filter(Boolean);
```

Reply:
Yeah I see you did that in https://github.com/18F/identity-idp/pull/4031/files

Reply:
Yep, I started drifting in favor of your suggestion once I'd come to a similar need in #4031. One or the other of these branches will need to have their conflicts resolved. I'll plan to move in that direction though.

Reply:
Updated in ce76ee9.